### DISCUSSION TITLE
how to use returning clause

### DISCUSSION THREAD

**Discussion by oryaacov** [aarondl/sqlboiler#988] [Q&A]
Hello, 
I wish to know how I can use postgres `returning` clause with SQL Boiler elegantly.

For example, I have the following query which update a user and returning the user ID:

 ```
err = queries.RawG(`update users
                     set name='test-name'
                     WHERE email='test@example.com'
                     RETURNING id`).Bind(ctx, boil.GetDB(), &user)
```

The following is working as expected, the user has been updated and I got back the user id. 
but I'm looking for more elegant ways to do the same without using `queries.RawG`

**Comment by aarondl** [comment]
I don't think we have any such ability since returning isn't actually a real concept in some of our lowest common denominator drivers (*cough* mysql *cough*) which would make this into 2 queries. I'd be opened to improvements in ergonomics around this for postgres specific use cases. Perhaps like we do upsert, we could have a special returning function that does some sort of magic.
